src/claw.cpp

```diff
@@ -0,0 +1,105 @@
+#include "claw.hpp"
+
+void Claw::open() {
+    uartComm << "#n M2232 V0\n";
+}
+
+void Claw::openUntilReleased() {
+    hwlib::cout << "Opening the claw until an item has been detected..." << hwlib::endlRet;
+
+    /// Move the gripper motor until item is detected
+    hwlib::wait_ms(1000);
+    hwlib::cout << "Item detected!" << hwlib::endlRet;
+    /// Stop the gripper motor
+
+    /// Set the current position of the motor (if position, depends on if the uArm Swift Pro has an actual actuator)
+    position = 23;
+
+    hwlib::cout << "Stopping gripper motor..." << hwlib::endlRet;
+}
+
+void Claw::close() {
+    uartComm << "#n M2232 V1\n";
+}
+
+void Claw::closeUntilGrabbed() {
+    hwlib::cout << "Closing the claw until an item has been detected..." << hwlib::endlRet;
+
+    /// Move the gripper motor until item is detected
+    hwlib::wait_ms(1000);
+    hwlib::cout << "Item detected!" << hwlib::endlRet;
+    /// Stop the gripper motor
+
+    /// Set the current position of the motor (if position, depends on if the uArm Swift Pro has an actual actuator)
+    position = 55;
+
+    hwlib::cout << "Stopping gripper motor..." << hwlib::endlRet;
+}
+
+void Claw::incrementAngle() {
+    /// Increment the angle of the gripper with the finest precision possible.
+    if (position >= 100)
+        return;
+
+    hwlib::cout << "Incrementing angle..." << hwlib::endlRet;
+
+    /// Increment angle by one step
+    position++;
+}
+
+void Claw::decrementAngle() {
+    /// Decrement the angle of the gripper with the finest precision possible.
+
+    if (position <= 0)
+        return;
+
+    hwlib::cout << "Incrementing angle..." << hwlib::endlRet;
+
+    /// Increment angle by one step
+    position--;
+}
+
+bool Claw::isOpen() {
+    return (position > 0) ? true : false;
+}
+
+bool Claw::isClosed() {
+    return (position == 0) ? true : false;
+}
+
+void Claw::setPosition(unsigned int destPos) {
+    /// Check if the destination position is in range and not in the current position
+    if (position <= 100 && destPos != position) {
+        /// Move the motors to the specified position
+
+        /// Set the new position
+        position = destPos;
+    }
+}
+
+unsigned int Claw::getPosition() {
+    return position;
+}
+
+char *Claw::getUarmFirmwareVersion(char response[15]) {
```

Comment:
Als het mogelijk is, probeer gebruik te maken van de C++ array.

Reply:
Good point. Is it OK for you that we will change this within the latest codebase ([the feature-uart-gripper](https://github.com/R2D2-2018/ROBOT_CLAW/tree/feature-uart-gripper) branch)?

Reply:
No, the point of these reviews is to make sure only code that is up to the right standards are in the master. So even if it's already changed in some other branch, it should also be part of the pull-request.  
If you already have new commits in the latest branch you can just make a new temporary branch for pull-request fixes.

Reply:
Ok, I just tested you the use of a C++ array. Unfortunately, in this function it cannot been (easly) done. As this method ```getUarmFirmwareVersion(char response[15])``` is used for printing the current firmware version located on the uArm, it uses ```hwlib::cout```. As hwlib::cout does not have an operator for a ```std::array``` or ```std::string``` this is not possible. However we could write it by ourselves, but in my opinion this would create unwanted overhead. In our case, an char[] style array would be the best fit.

Reply:
If there is a good reason for using a char[] array (like this), then it's fine.